Packages/com.unity.inputsystem/InputSystem/Devices/NoiseFilter.cs

```diff
@@ -0,0 +1,246 @@
+using System;
+using System.Collections.Generic;
+using UnityEngine.Experimental.Input.LowLevel;
+using UnityEngine.Experimental.Input.Processors;
+using UnityEngine.Experimental.Input.Utilities;
+
+namespace UnityEngine.Experimental.Input
+{
+    /// <summary>
+    /// A filter for individual devices to check if events or device states contain significant, relevant changes.
+    /// Irrelevant changes are any updates controls that are tagged as 'noisy', and any state change that turns into a no-operation change once processors are applied (e.g. in deadzone joystick movements).
+    /// </summary>
+    public class NoiseFilter
+    {
+        /// <summary>
+        /// Simple cached value identifying how to filter this element (Bitmask, or individual type filtered).
+        /// </summary>
+        public enum EElementType
+        {
+            TypeUnknown = 0,
+            TypeFull,
```

Comment:
`Full` as a name isn't very clear to me. Can't really tell the difference between `full` vs `float` etc.